src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -199,19 +199,38 @@ public function hasTable($tableName)
             return true;
         }
 
+        if (strpos($tableName, '.') !== false) {
+            list($schema, $table) = explode('.', $tableName);
+            $exists = $this->queryTableExists($schema, $table);
+            // Only break here on success, because it is possible for table names to contain a dot.
+            if (!empty($exists)) {
```

Comment:
Is this a boolean? Then the empty() is a code smell here.

Reply:
You didn't remark this on his old code :P

So I didn't really bother to change all that, but it seems the queryTableExists actually returns the table name... which is indeed a bit confusing. I'll clean it up a bit.